packages/utils/src/userop.ts

```diff
@@ -1,51 +1,131 @@
 
         if (rcp.status === "success") {
-            // find if any logs are UserOperationEvent
+            // find if any logs are UserOperationEvent or AccountDeployed
             const r = rcp.logs
                 .map((l) => {
                     if (l.address === rcp.to) {
                         try {
-                            const log = decodeEventLog({ abi: EntryPointAbi, data: l.data, topics: l.topics })
+                            const log = decodeEventLog({
+                                abi: EntryPointAbi,
+                                data: l.data,
+                                topics: l.topics
+                            })
+                            if (log.eventName === "AccountDeployed") {
+                                return {
+                                    userOperationHash: log.args.userOpHash,
+                                    success: !!log.args.factory,
+                                    accountDeployed: true
+                                }
+                            }
                             if (log.eventName === "UserOperationEvent") {
-                                return log
-                            } else {
-                                return undefined
+                                return {
+                                    userOperationHash: log.args.userOpHash,
+                                    success: !!log.args.success,
+                                    accountDeployed: false
+                                }
                             }
+                            return undefined
                         } catch (_e) {
                             sentry.captureException(_e)
-                            console.log("error decoding transaction inclusion status", _e)
                             return undefined
                         }
-                    } else {
-                        return undefined
                     }
+                    return undefined
                 })
-                .find((l) => l !== undefined)
+                .reduce(
+                    (
+                        result: {
+                            [userOperationHash: HexData32]: {
+                                userOperationHash: HexData32
+                                accountDeployed: boolean
+                                success: boolean
+                            }
+                        },
+                        log
+                    ) => {
+                        if (log) {
+                            return {
+                                [log.userOperationHash]: {
+                                    accountDeployed:
+                                        log.accountDeployed ||
+                                        result[log.userOperationHash]
+                                            ?.accountDeployed,
+                                    success:
+                                        log.success ||
```

Comment:
not &&?

Reply:
this is old code and I have no idea what it does 😅 I just formatted it.